parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/ParallelEoSStreamProcessor.java

```diff
@@ -835,6 +844,31 @@ private void processWorkCompleteMailBox() {
         }
     }
 
+    @Override
+    public void pollBatch(Consumer<List<ConsumerRecord<K, V>>> usersVoidConsumptionFunction) {
+        validateBatch();
+
+        Function<List<ConsumerRecord<K, V>>, List<Object>> wrappedUserFunc = (recordList) -> {
+            log.trace("asyncPoll - Consumed set of records ({}), executing void function...", recordList.size());
+            usersVoidConsumptionFunction.accept(recordList);
+            return UniLists.of(); // user function returns no produce records, so we satisfy our api
+        };
+        Consumer<Object> voidCallBack = (ignore) -> log.trace("Void callback applied.");
+        supervisorLoop(wrappedUserFunc, voidCallBack);
+    }
+
+    private void validateBatch() {
+        if (!options.getBatchSize().isPresent()) {
+            throw new IllegalArgumentException("Using batching function, but no batch size specified in options");
```

Comment:
Perhaps this message should be a bit clearer and refer to ParallelConsumerOptions explicitly? For new users coming from kafka-consumer, `pollBatch` would be the default/natural way to process messages as kafkaConsumer.poll is only batch-based. Most users don't read docs very carefully, so this exception will probably be the first encounter with PC for many users, and batch size might be confused with `max.poll.records` etc.